### ISSUE TITLE
How to remove middleware, such as ImageSharp,  in OWIN

### ISSUE THREAD

**Issue by infofromca** [OrchardCMS/OrchardCore#1477]
In my own module, I want to implement my own services.AddImageSharpCore  for ImageSharp.
1---in this way, I need to remove it from OrchardCore.Media module, I think.
2---how?, can orchard do that? it seems that asp.net core can not do that directly.
thanks
andy

**Comment by sebastienros** [comment]
Two solutions:
- Make the middleware extensible using options to suit your needs
- Create a new feature for it so it can be disabled if needed

We could also do both.

However you can already configure it from your module, you can call `AddImageSharpCore` a second time and override anything you want. We tried it already to customize the compression levels. 

c.f. https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Media/Startup.cs#L85

**Comment by infofromca** [comment]
overriding is not the best choice.
my situation is that in my special situation(module), i do not want to apply 
public static int[] Sizes = new[] { 16, 32, 50, 100, 160, 240, 480, 600, 1024, 2048 };
because editing image is inside admin panel.
but I also want to keep applying
public static int[] Sizes = new[] { 16, 32, 50, 100, 160, 240, 480, 600, 1024, 2048 };
when in normal resize -- external url.

**Comment by sebastienros** [comment]
> because editing image is inside admin panel.

I don't understand this statement. Can you rephrase what you are doing?

**Comment by infofromca** [comment]
> i do not want to apply
> public static int[] Sizes = new[] { 16, 32, 50, 100, 160, 240, 480, 600, 1024, 2048 };
> because editing image is inside admin panel.

means

>  i do not want to apply
> public static int[] Sizes = new[] { 16, 32, 50, 100, 160, 240, 480, 600, 1024, 2048 };
> in the dashboard for editing image .

==================
there are 2 situation. 
1- use my custom feature
2--keep use the original  feature of OrchardCore.Media module

**Comment by sebastienros** [comment]
You can customize these settings now.
